### ISSUE TITLE
error: "Disk space is low!", ignoring disk space limitations

### ISSUE THREAD

**Issue by nikitasius** [bitcoinknots/bitcoin#8]
### Describe the issue
Program ignore disk space limitation, what i set up on start

#### If so, please list the steps to reproduce below:
1. start bitcoin-0.14.2/bin/bitcoin-qt
2. select checkbox what you want to limit data to 1024Mb (~6 days)
3. bitcoin use all disk space (~16Gb in my case) and throws an error.

### Expected behaviour
Program syncing a blockchain using only 1Gb of free space on my hdd

### Actual behaviour
Even if i selected to limit disk space to 1024Mb, program used all available space. folder `.bitcoin/blocks/`have size of 16Gb.

### What version of Bitcoin Knots are you using?
`bitcoin-0.14.2.knots20170618-x86_64-linux-gnu.tar.gz`

### Machine specs:
- OS: debian 8.8/4.9.0-0.bpo.2-amd64 #1 SMP Debian 4.9.18-1~bpo8+1 (2017-04-10) x86_64 GNU/Linux
- CPU: i5-i5200
- RAM: 8Gb (2x4Gb exact modules, i.e. model)
- Disk size: 1Tb
- Disk Type (HD/SDD): hdd

### Any extra information that might be useful in the debugging process.

> 2017-07-06 05:53:20 UpdateTip: new best=00000000000000006abd2d638e7b532a9b5938af1449666dead4694b630d2cfa height=278269 version=0x00000002 log2_work=75.356384 tx=30280869 date='2014-01-02 11:55:20' progress=0.128740 cache=318.1MiB(491213tx)
> 2017-07-06 05:53:20 UpdateTip: new best=00000000000000017bac9a8e85660ad348050c789922d5f8fe544d473368be1a height=278270 version=0x00000002 log2_work=75.356566 tx=30281382 date='2014-01-02 12:35:05' progress=0.128742 cache=318.2MiB(491336tx)
> 2017-07-06 05:53:20 *** Disk space is low!
> 2017-07-06 05:53:30 GUI: QXcbConnection: XCB error: 3 (BadWindow), sequence: 17135, resource id: 19042778, major code: 40 (TranslateCoords), minor code: 0
> 2017-07-06 05:53:30 ERROR: ProcessNewBlock: ActivateBestChain failed
> 2017-07-06 05:53:30 tor: Thread interrupt
> 2017-07-06 05:53:30 net thread exit
> 2017-07-06 05:53:30 torcontrol thread exit
> 2017-07-06 05:53:30 opencon thread exit
> 2017-07-06 05:53:30 addcon thread exit
> 2017-07-06 05:53:31 *** Disk space is low!
> 2017-07-06 05:53:31 scheduler thread interrupt
> 2017-07-06 05:53:31 Shutdown: In progress...
> 2017-07-06 05:53:32 GUI: QXcbConnection: XCB error: 3 (BadWindow), sequence: 17327, resource id: 19043178, major code: 40 (TranslateCoords), minor code: 0
> 2017-07-06 05:53:32 ERROR: ProcessNewBlock: ActivateBestChain failed
> 2017-07-06 05:53:32 msghand thread exit
> 2017-07-06 05:53:32 Dumped mempool: 3e-06s to copy, 0.085057s to dump
> 2017-07-06 05:53:32 *** Disk space is low!
> 2017-07-06 05:53:33 Shutdown: done

**Comment by luke-jr** [comment]
Does a `bitcoin_rw.conf` file exist? Can you post its contents?

**Comment by nikitasius** [comment]
> Does a bitcoin_rw.conf file exist? Can you post its contents?

I wiped all (software, config, .bitcoin folder). I will try it again and i will tell you resultats.

**Comment by nikitasius** [comment]
Well..

error: http://i.imgur.com/BzvpCWr.png
blocks folder: http://i.imgur.com/KVswpaQ.png
before: `/dev/sda2                    717G  665G   16G  98% /`
after: `/dev/sda2                    717G  681G   51M 100% /`

Settings on start: http://i.imgur.com/5NJUVfq.png

Thats how i start it: `~/patatipatata/0_btc/bitcoin-0.14.2/bin$ ./bitcoin-qt`

`patatipatata` - ecryptfs folder.

There is no `bitcoin_rw.conf` file (not in `.bitcoin` folder and not in `~/.config/Bitcoin`) there is only `Bitcoin-Qt.conf` file: 
[Bitcoin-Qt.conf.txt](https://github.com/bitcoinknots/bitcoin/files/1128714/Bitcoin-Qt.conf.txt)

**Comment by nikitasius** [comment]
i got some free space and..
before `/dev/sda2                    717G  544G  137G  80% /`
after (not finished yet) `/dev/sda2                    717G  586G   95G  87% /`
Looks like this software really don't care about "use 1Gb or 4Gb and not more setting".
![screenshot - 07072017 - 07 48 09 am](https://user-images.githubusercontent.com/3670331/27944710-04c0d364-62e9-11e7-996d-56e4f60cde7d.png)

**Comment by nikitasius** [comment]
well well well. idk what the fuck with this software
![screenshot - 09272017 - 09 14 01 pm](https://user-images.githubusercontent.com/3670331/30932787-d3b75aa6-a3c8-11e7-9632-3082dfb29c66.png)

You select 2048 Mb, it continue and continue download. Damn i no need this 150 Gb of blockchains. 

Since 7 jul no info -.-

**Comment by luke-jr** [comment]
Check `~/Library/Application Support/Bitcoin/`

**Comment by nikitasius** [comment]
I dont have such folder
```
nikitas@pentagon:~/Downloads$ cd ~/Library/Application Support/Bitcoin/
bash: cd: /home/nikitas/Library/Application: No such file or directory
```
btw reached 5.1Gb
![screenshot - 09272017 - 09 35 17 pm](https://user-images.githubusercontent.com/3670331/30933780-d271f900-a3cb-11e7-858a-4f96ceb3630a.png)

**Comment by nikitasius** [comment]
here is file what was running now https://bitcoinknots.org/files/0.15.x/0.15.0.knots20170914/bitcoin-0.15.0.knots20170914-x86_64-linux-gnu.tar.gz

**Comment by luke-jr** [comment]
Sorry, your theme made me think you were using Mac.

On Linux, the directory is `~/.bitcoin`

There should be a `bitcoin_rw.conf` file in it...

**Comment by nikitasius** [comment]
well..


```
nikitas@pentagon:~$ ls ~/.bitcoin
ls: cannot access /home/nikitas/.bitcoin: No such file or directory
nikitas@pentagon:~$ ls ~/.config/Bitcoin/
Bitcoin-Qt.conf
nikitas@pentagon:~$ ls ~/patatipatata/0_btc/.bitcoin/
banlist.dat  chainstate  debug.log          mempool.dat  wallet.dat
blocks       db.log      fee_estimates.dat  peers.dat
nikitas@pentagon:~$ 
```

**Comment by luke-jr** [comment]
That's odd. Are you setting `$HOME` or `-datadir` when you run Bitcoin Knots? Why is it using `~/patatipatata/0_btc/` for the `.bitcoin` directory?

**Comment by nikitasius** [comment]
nope, i should?

clean run:

1. i downloaded https://bitcoinknots.org/files/0.15.x/0.15.0.knots20170914/bitcoin-0.15.0.knots20170914-x86_64-linux-gnu.tar.gz
2. i unpacked it into ~/patatipatata/0_btc/
3. i open terminal in ~/patatipatata/0_btc/bitcoin-0.15.0/bin
4. i run ./bitcoin-qt

next i setup as 
![screenshot - 09282017 - 08 30 21 am](https://user-images.githubusercontent.com/3670331/30952210-2f626202-a427-11e7-963d-a0c715f26d79.png)

and some time after i have
![screenshot - 09282017 - 09 07 14 am](https://user-images.githubusercontent.com/3670331/30953438-39207270-a42c-11e7-990c-e3cdde977b04.png)

patatipatata - ecryptfs area.

**Comment by nikitasius** [comment]
still increasing
![screenshot - 09282017 - 09 14 55 am](https://user-images.githubusercontent.com/3670331/30953701-4f764cc4-a42d-11e7-9e07-0cb0c1ff7126.png)

**Comment by luke-jr** [comment]
Reproduced the bug, will fix soon.

**Comment by luke-jr** [comment]
Please confirm that it is fixed in http://luke.dashjr.org/programs/bitcoin/files/bitcoind/misc/0.15.x/tmp-942fcf8e60/

**Comment by nikitasius** [comment]
Looks fixed!

Before: https://github.com/bitcoinknots/bitcoin/issues/8#issuecomment-332626483

Now:
![screenshot - 09302017 - 08 39 29 pm](https://user-images.githubusercontent.com/3670331/31048535-3945fc8a-a61f-11e7-8035-5c1ce66be789.png)


and here are settings:
![screenshot - 09302017 - 08 13 45 pm](https://user-images.githubusercontent.com/3670331/31048539-4b4176d0-a61f-11e7-98de-b51e9a8418c3.png)


I will let it download all. Great job. Finally i will be able to use bitcoin API <3

**Comment by nikitasius** [comment]
what finally you did? And why debian was affected?

**Comment by luke-jr** [comment]
Had to reorganize the code a bit. It was trying to set the prune value before it determined the datadir. So I made it wait to write it until later.

**Comment by luke-jr** [comment]
Final fix will be released in 0.15.1. Until then, the custom build is needed for the first-time run only - once it's set correctly, you can use the normal 0.15.0 release.
